apps/music/js/ui/views/player_view.js

```diff
@@ -826,103 +832,107 @@ var PlayerView = {
+            // until the activity returns. Then we pass the name of this magic
+            // setting as a secret undocumented property of the activity so that
+            // the ringtones app can use it.
+            //
+            // This done as much as possible in a self-invoking function to make
+            // it easier to remove the hack when we have a real bug fix.
+            //
+            // See also the corresponding code in apps/ringtones/js/share.js
+            //
+            // HACK HACK HACK
+            (function() {
+              // This are the magic names we'll use for this hack
+              var hack_activity_property = '_hack_hack_shut_up';
+              var hack_setting_property = 'music._hack.pause_please';
+
+              // Listen for changes to the magic setting
+              navigator.mozSettings.addObserver(hack_setting_property,
+                                                observer);
+
+              // Pass the magic setting name as part of the activity request
+              activityData[hack_activity_property] = hack_setting_property;
+
+              // Now initiate the activity. This code is the same as the
+              // normal non-hack code in the if clause above.
+              var a = new MozActivity({
+                name: 'share',
+                data: activityData
+              });
+
+              a.onerror = a.onsuccess = cleanup;
+
+              // This is the function that pauses the music if the activity
+              // handler sets the magic settings property.
+              function observer(e) {
+                // If the value of the setting has changed, then we pause the
+                // music. Note that we don't care what the new value of the
+                // setting is.  We only care whether it has changed. The
+                // ringtones app will just toggle it back and forth between
+                // true and false.
+                PlayerView.pause();
+              }
 
-            // When the activity is done, we stop observing the setting.
-            // And if we have been paused, then we resume playing.
-            function cleanup() {
-              navigator.mozSettings.removeObserver(hack_setting_property,
-                                                   observer);
-              if (PlayerView.playStatus === PLAYSTATUS_PAUSED) {
-                PlayerView.play();
+              // When the activity is done, we stop observing the setting.
+              // And if we have been paused, then we resume playing.
+              function cleanup() {
+                navigator.mozSettings.removeObserver(hack_setting_property,
+                                                     observer);
+                if (PlayerView.playStatus === PLAYSTATUS_PAUSED) {
+                  PlayerView.play();
+                }
               }
-            }
-          }());
-        }
+            }());
+          }
+        });
```

Comment:
So I'm assuming that none of this share code has changed, and that all you've done is insert the lazyloading step which changes the indentation of everything else.